datatypes/src/main/java/org/hyperledger/besu/datatypes/CallParameter.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright contributors to Besu.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.datatypes;
+
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalLong;
+
+import org.apache.tuweni.bytes.Bytes;
+
+/**
+ * Interface defining parameters for Ethereum transaction calls and simulations.
+ *
+ * <p>This interface provides access to all the parameters that can be specified when making calls
+ * to the Ethereum network, including both legacy and EIP-1559 transaction parameters, as well as
+ * EIP-4844 blob transaction parameters.
+ *
+ * <p>All parameters are optional to support various call scenarios where only a subset of
+ * parameters may be specified.
+ */
+public interface CallParameter {
```

Comment:
I don't see another implementation other than ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/CallParameter.java so not sure to understand why is this introduced ?